openlibrary/plugins/openlibrary/js/realtime_account_validation.js

```diff
@@ -90,14 +100,30 @@ export function initRealTimeValidation() {
     }
 
     function validatePassword() {
-        const value_password = $(this).val();
+        const value_password = $('#password').val();
         if (value_password !== '' && (value_password.length < 3 || value_password.length > 20)) {
             renderError('#password', '#passwordMessage', i18nStrings['password_length_err']);
         }
         else clearError('#password', '#passwordMessage');
     }
 
-    $('#username').on('blur', validateUsername);
-    $('#emailAddr').on('blur', validateEmail);
-    $('#password').on('blur', validatePassword);
+    function validateInput(input) {
+        const id = $(input).attr('id');
+        const validate = validationFns[id];
```

Comment:
Rebecca and I chatted, and we think have this use if statement since there are only three (and maybe throw an error if it's not one of the three!) will remove some of the indirection here and make the code easier to follow.